lib/datasource.js

```diff
@@ -632,6 +632,7 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       if (idProp.generated && forceId !== false) {
         forceId = true;
       }
+      settings.forceId = forceId;
```

Comment:
@bajtos I had written a comment on we can't forceId calculation and setting in model-builder. Looks like it was wiped out after I did a commit with force push option, after I had rebased my PR. 

If we set  forceId in model-builder.js instead of datasource.js, 60 tests in loopback are failing. Hence I left the forceId back the settings[] in datasource.js. I will sync up with you on this tomorrow.